types/index.d.ts

```diff
@@ -1,45 +1,98 @@
 // -------------------------------------------------------------------------
-// Welcome to the sketchbook of our library's type definitions. 
+// Welcome to the sketchbook of our library's type definitions.
 //--------------------------------------------------------------------------
 //
-// Like a promise of "we'll do it tomorrow," we've sketched out the outlines but the details? 
+// Like a promise of "we'll do it tomorrow," we've sketched out the outlines but the details?
 // Well, they will be defined later... This finger never eats but always says 'I'll pick the tab next time!'
 //
 // Feel adventurous? Help us fill in the gaps – don't wait for tomorrow!
 // -------------------------------------------------------------------------
 
-export class Agent {
+import { Tool } from "langchain/tools";
+import type { AGENT_STATUS_enum, TASK_STATUS_enum } from "./enums.d.ts";
+import type {
+  BaseAgent,
+  IBaseAgentParams,
+  ILLMConfig,
+  ITaskStats,
+  TAgentTypes,
+  TStore,
+} from "./types.d.ts";
 
-    [key: string]: any;
+declare module "agenticjs" {
+  export class Agent {
+    agentInstance: BaseAgent;
+    type: string;
 
-    constructor(props?: Record<string, any>) {
-        Object.assign(this, props);
-    }    
+    constructor(type: TAgentTypes, config: IBaseAgentParams);
 
-    // Implementation details will be defined later... 
-    // Yes, this is the 'I’ll clean my desk next week' kind of later.
-}
+    createAgent(type: TAgentTypes, config: IBaseAgentParams): BaseAgent;
+    executeTask(task: Task): Promise<any>;
 
-export class Task {
+    setStore(store: TStore): void;
+    setEnv(env: Record<string, any>): void;
+    setStatus(status: AGENT_STATUS_enum): void;
 
-    [key: string]: any;
+    id(): string;
+    name(): string;
+    role(): string;
+    goal(): string;
+    background(): string;
+    tools(): Tool[];
+    status(): AGENT_STATUS_enum;
+    llmConfig(): ILLMConfig;
+    llmSystemMessage(): string;
+    forceFinalAnswer(): boolean;
+  }
 
-    constructor(props?: Record<string, any>) {
-        Object.assign(this, props);
-    }    
+  export interface ITaskParams {
+    title?: string;
+    description: string;
+    expectedOutput: any;
+    agent: BaseAgent;
+    isDeliverable?: boolean;
+  }
 
-    // Implementation details will be defined later... 
-    // Like that gym membership we all get in January.
-}
+  export class Task {
+    id: string;
+    title: string;
+    description: string;
+    expectedOutput: string;
+    isDeliverable: boolean;
+    agent: BaseAgent;
+    status: TASK_STATUS_enum;
+    result: any; // ? Need more context
```

Comment:
It is the result of a given task. Once an Agent has finished working on it. It is mean to be a read only attribute.

Reply:
Can give an example or reference to what the type/structure of the response would be like?

Reply:
A string.